spirv_cross.cpp

```diff
@@ -1703,6 +1703,13 @@ void Compiler::parse(const Instruction &instruction)
 		}
 
 		auto &var = set<SPIRVariable>(id, type, storage, initializer);
+        
+        auto &ttype = get<SPIRType>(type);
```

Comment:
Does the input SPIR-V not have these decorations?

Reply:
not for NonReadable at least. since the texture reading/writing code resets those decorations it looks safe to set both at the beginning

Reply:
Hm ... Is this coming from HLSL by any chance? In GLSL there is writeonly/readonly which should cover these things automatically.

Reply:
yeah, all of our sources are hlsl